16_wordle/main.go

```diff
@@ -99,3 +99,88 @@ func checkChar(chosenWordSlice []string, inputSlice []string) []string {
 	}
 	return containedSlice
 }
+
+func checkCharPrecise(chosenWordSlice []string, inputSlice []string) []int {
+	var containedSlice []int
+	// 1 = contained but incorrect loc
+	// 2 = contained in correc loc
+	// 0 not contained
+
+	// correct word check
+	if slices.Equal(chosenWordSlice, inputSlice) {
+		return []int{2, 2, 2, 2, 2}
+	}
+
+	// for inIndex, inChar := range in {
+	//   for chosenIndex, chosenChar := range chosen {
+	//     if inChar != chosenChar{
+	//       // TODO
+	//     } else {
+	//       if inIndex == chosenIndex {
+	//         // TODO
+	//       } else {
+	//         // TODO
+	//       }
+	//     }
+	//   }
+	// }
+
+	for inputIndex, inputValue := range inputSlice {
```

Comment:
Here you have 3 cases:

- 2: same value, same index, you can get that with `inputValue == chosenValue && inputIndex == chosenIndex`
- 1: the value is in the chosen word list
- 0: the value is not in the chosen word list

Maybe you don't need a double loop but there's a different way to approach it?